src/include/execution/sql/runtime_types.h

```diff
@@ -415,6 +415,11 @@ class EXPORT Decimal {
   /** Underlying native data type. */
   using NativeType = T;
 
+  /**
+   * Empty constructor
+   */
+  Decimal() = default;
```

Comment:
For the person reviewing this, the reason that I added this constructor is because `TopKElements` has a `min_key_` field that gets default constructed. https://github.com/cmu-db/noisepage/blob/1fdb71c57714260c1cc8105c072b365f46855209/src/include/optimizer/statistics/top_k_elements.h#L348-L352
Honestly it's a little suspect because it doesn't seem like there's any logic in that class to check if `min_key_` has been properly initialized or not, but it's an existing class with existing tests so I honestly didn't look too much into it.

Reply:
Thanks for calling this out. It definitely is concerning... The thing making it even a little more difficult to deal with is the default constructed `Decimal` is just going to have it's underlying value defaulted to `0`, so it is indistinguishable from an instance that was constructed with the non-default constructor and a `0` value.

I can't think of a good solution off the top of my head, but something we might be able to add soon once C++20 concepts are stabilized on both GCC and Clang is to do a little concepts-based polymorphism where we overload the constructor for `TopKElements` based on whether the `KeyType` is default constructible (already in `<type_traits>` I believe) and in the event that it is not require that an additional initializer be provided to the constructor.

Reply:
The other thing making me feel slightly better about this is that all the other runtime types have equally useless default constructors.